typescript/src/schema/proto.ts

```diff
@@ -744,6 +753,69 @@ function augmentMethod(
   return method;
 }
 
+/* Interface and method to grab selective gapic methods from the service yaml
+along with other potential selective gapic config options. */
+interface SelectiveGapicConfig {
+  isSelectiveGapic: boolean;
+  generateOmittedAsInternal: boolean | undefined;
+  methods: string[];
+}
+
+export function getSelectiveGapic(
+  serviceYaml: ServiceYaml | undefined
+): SelectiveGapicConfig {
+  const selectiveGapicConfig =
+    serviceYaml?.publishing?.library_settings?.[0]?.typescript_settings?.common
+      ?.selective_gapic_generation;
+
+  const methods: string[] = [];
+  const generateOmittedAsInternal =
+    selectiveGapicConfig?.generate_omitted_as_internal ?? false;
+  const isSelectiveGapic = !!selectiveGapicConfig;
+
+  // If selective gapic is set, then we can parse the methods from the config.
+  if (selectiveGapicConfig?.methods) {
+    // We find the method name and add it to methods array.
+    // Example: `google.cloud.bigquery.v2.CancelJobRequest` becomes `CancelJobRequest`.
+    for (const method of selectiveGapicConfig.methods) {
+      const lastDotIndex = method.lastIndexOf('.');
+      if (lastDotIndex !== -1) {
+        methods.push(method.substring(lastDotIndex + 1));
+      }
+    }
+  }
+
+  return {
+    isSelectiveGapic,
+    generateOmittedAsInternal,
+    methods,
+  };
+}
+
+enum SelectiveGapicType {
+  PUBLIC = 'public',
+  HIDDEN = 'hidden',
+  INTERNAL = 'internal',
+}
+
+export function isMethodSelectiveGapic(
```

Comment:
Very nitpicky, but it seems like `is` implies boolean, so maybe it should be something like `selectiveGapicMethodType`... not sure if that's a good name either :D